Modules/Devices/src/Devices.Infrastructure/PushNotifications/DirectPush/DirectPushService.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Backbone.Modules.Devices.Application.Infrastructure.PushNotifications;
+using Backbone.Modules.Devices.Domain.Aggregates.PushNotifications.Handles;
+using Enmeshed.DevelopmentKit.Identity.ValueObjects;
+using Microsoft.Extensions.Logging;
+
+namespace Backbone.Modules.Devices.Infrastructure.PushNotifications.DirectPush;
+public class DirectPushService : IPushService
+{
+    private readonly IPnsRegistrationRepository _pnsRegistrationRepository;
+    private readonly ILogger<DirectPushService> _logger;
+    private readonly CancellationToken _cancellationToken;
+
+    public DirectPushService(IPnsRegistrationRepository pnsRegistrationRepository, ILogger<DirectPushService> logger, CancellationToken cancellationToken)
```

Comment:
Let me guess: you have no idea what a CancellationToken is. Otherwise you would never get the idea to inject one in the constructor.